includes/Modules/Analytics.php

```diff
@@ -1331,4 +1365,97 @@ public function check_service_entity_access() {
 		return true;
 	}
 
+	/**
+	 * Validates the report metrics.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $metrics The metrics to validate.
+	 *
+	 * @return null|WP_Error NULL if metrics are valid, otherwise WP_Error.
+	 */
+	protected function validate_report_metrics( $metrics ) {
+		if ( false === $this->is_using_shared_credentials ) {
+			return null;
+		}
+
+		$valid_metrics = apply_filters(
+			'googlesitekit_shareable_analytics_metrics',
+			array(
+				'ga:sessions',
+				'ga:users',
+				'ga:pageviews',
+				'ga:uniquePageviews',
+				'ga:bounceRate',
+				'ga:avgSessionDuration',
+				'ga:adsenseRevenue',
+				'ga:adsenseECPM',
+				'ga:adsensePageImpressions',
+				'ga:goalCompletionsAll',
+			)
+		);
+
+		$invalid_metrics = array_diff( $metrics, $valid_metrics );
+
+		if ( count( $invalid_metrics ) > 0 ) {
+			return sprintf(
+				/* translators: %s is replaced with a comma separated list of the invalid metrics. */
+				_n(
+					'Unsupported metric requested: %s',
+					'Unsupported metrics requested: %s',
+					count( $invalid_metrics ),
+					'google-site-kit'
+				),
+				implode( ', ', $invalid_metrics )
+			);
+		}
+
+		return null;
+	}
+
+	/**
+	 * Validates the report dimensions.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $dimensions The dimensions to validate.
+	 *
+	 * @return null|WP_Error NULL if dimensions are valid, otherwise WP_Error.
+	 */
+	protected function validate_report_dimensions( $dimensions ) {
```

Comment:
It would be great to get some test coverage on these methods, but I see how that could be a bit problematic as-is. Maybe we could extract these into validation-specific classes? The class could be specific to validating shareable report dimensions so we could unit test it pretty easily and leave the `is_using_shared_credentials` bit to control whether or not such validation is applied or not.

Then we might add only a test or two to the Analytics class to make sure it is wired-up as expected.

Reply:
Created #5728 to address this. Thank you!